roboteam_ai/src/conditions/CanShootToTarget.cpp

```diff
@@ -0,0 +1,35 @@
+#include "CanShootToTarget.h"
+
+namespace rtt {
+namespace ai {
+
+CanShootToTarget::CanShootToTarget(std::string name, bt::Blackboard::Ptr blackboard)
+        :Condition(std::move(name), std::move(blackboard)) { };
+
+bt::Node::Status CanShootToTarget::onUpdate() {
+
+    // get the target position to which we want a clear line
+    if (properties->hasVector2("target")) {
+		target = properties->getVector2("target");
+	} else {
+		return bt::Node::Status::Failure;
+	}
+
+    // get the margin, which is the min distance from the line
+	if (properties->hasDouble("margin")) {
+		margin = properties->getDouble("margin");
+	} else {
+		margin = MAX_DIST_FROM_LINE;
+	}
+
+	bool canShoot = control::ControlUtils::clearLine(robot->pos, target, world::world->getWorld(), margin, true);
```

Comment:
clearLine is a vague name for a function, on it's own it would mean that you 'clear' the line i.e. delete/reset it.